pkg/cmd/cli/backuplocation/set.go

```diff
@@ -104,36 +105,40 @@ func (o *SetOptions) Run(c *cobra.Command, f client.Factory) error {
 		}
 	}
 
-	location := &velerov1api.BackupStorageLocation{}
-	err = kbClient.Get(context.Background(), kbclient.ObjectKey{
-		Namespace: f.Namespace(),
-		Name:      o.Name,
-	}, location)
+	defalutBSLs, err := storage.GetDefaultBackupStorageLocations(context.Background(), kbClient, f.Namespace())
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	if o.DefaultBackupStorageLocation {
-		// There is one and only one default backup storage location.
-		// Disable the origin default backup storage location.
-		locations := new(velerov1api.BackupStorageLocationList)
-		if err := kbClient.List(context.Background(), locations, &kbclient.ListOptions{Namespace: f.Namespace()}); err != nil {
-			return errors.WithStack(err)
+	if !o.DefaultBackupStorageLocation { // unset default backup storage location
```

Comment:
If the `velero backup-location set` command doesn't set the `--default` parameter, it can also hit this logic.
I think it can confuse the user because the user could meet the default BSL validation error when the user doesn't have the intention to set the default BSL.

Reply:
you are right, I've modified it, please help me to check it again